man/systemd.unit.xml

```diff
@@ -976,59 +976,79 @@
+
+        <para>These configuration options are particularly useful in conjunction with the service setting
         <varname>Restart=</varname> (see
-        <citerefentry><refentrytitle>systemd.service</refentrytitle><manvolnum>5</manvolnum></citerefentry>); however,
-        they apply to all kinds of starts (including manual), not just those triggered by the
-        <varname>Restart=</varname> logic. Note that units which are configured for <varname>Restart=</varname> and
-        which reach the start limit are not attempted to be restarted anymore; however, they may still be restarted
-        manually at a later point, after the <replaceable>interval</replaceable> has passed.  From this point on, the
-        restart logic is activated again. Note that <command>systemctl reset-failed</command> will cause the restart
-        rate counter for a service to be flushed, which is useful if the administrator wants to manually start a unit
-        and the start limit interferes with that. Note that this rate-limiting is enforced after any unit condition
-        checks are executed, and hence unit activations with failing conditions do not count towards this rate
-        limit. This setting does not apply to slice, target, device, and scope units, since they are unit types whose
-        activation may either never fail, or may succeed only a single time.</para>
-
-        <para>When a unit is unloaded due to the garbage collection logic (see above) its rate limit counters are
-        flushed out too. This means that configuring start rate limiting for a unit that is not referenced continuously
-        has no effect.</para></listitem>
+        <citerefentry><refentrytitle>systemd.service</refentrytitle><manvolnum>5</manvolnum></citerefentry>);
+        however, they apply to all kinds of starts (including manual), not just those triggered by the
+        <varname>Restart=</varname> logic.</para>
+
+        <para>Note that units which are configured for <varname>Restart=</varname>, and which reach the start
+        limit are not attempted to be restarted anymore; however, they may still be restarted manually at a
+        later point, after the <replaceable>interval</replaceable> has passed. From that point on, the
+        restart logic is activated again. <command>systemctl reset-failed</command> will cause the restart
+        rate counter for a service to be flushed, which is useful if the administrator wants to manually
+        start a unit and the start limit interferes with that. Rate-limiting is enforced after any unit
+        condition checks are executed, and hence unit activations with failing conditions do not count
+        towards the rate limit.</para>
+
+        <para>When a unit is unloaded due to the garbage collection logic (see above) its rate limit counters
+        are flushed out too. This means that configuring start rate limiting for a unit that is not
+        referenced continuously has no effect.</para>
```

Comment:
this is something we should address. i.e. not GC unit that have "open ratelimiting time windows" if you grok what i mean. It's a bit messy though, since we probably need to protect things so that super-long timelimit windows won't mean GC is turned off entirely

Reply:
Yeah. I guess that this was added because behaviour can be surprising. So we should keep this until the improved code is merged.